### ISSUE TITLE
Always showing 0° temperature and 0% fan

### ISSUE THREAD

**Issue by lorenzosu** [tuxedocomputers/tuxedo-control-center#59]
Running version 1.0.7 which I got via an update on Manjaro. Kernel is 5.8.18-1
Laptop is a Book BC1507

As per title the control center is always showing a temperature of 0° and a fan at 0% which is clearly not possible.

**Comment by lorenzosu** [comment]
Update: I think that with this new version 1.0.7 (and maybe 0.1.7 of tuxedo-cc-wmi) is simply not working on my machine in fact the fan is going like crazy even with the CPU is near to idle.
Downgrading Tuxedo Control Center to 1.0.4 and tuxedo-cc-wmi to 0.1.6 everything seems to work as expected

**Comment by Matombo** [comment]
I checked the Manjaro package of tuxedo-cc-wmi and it has linux-headers only as an optional dependency. Without the correct linux-headers installed cpu and fan controll will not work with tuxedo-cc-wmi which is the interface used internally by the Tuxedo Controll Center.

Can you make sure that the correct linux-headers for your current kernel are installed and try again?

**Comment by vinzv** [comment]
Manjaro support is done at Manjaro. I opened an issue at their PKGBUILD for tuxedo-cc-wmi: https://gitlab.manjaro.org/packages/extra/tuxedo-cc-wmi/-/issues/1
Please track the problem over there as they can help you better. Thanks!

**Comment by vinzv** [comment]
Just a quick update @lorenzosu: according to Manjaro Git the dependencies should be fixed. If you run into any other inconvenience please do let us know.

**Comment by lorenzosu** [comment]
@Matombo and @vinzv thanks for the very quick feedback. I now had time to test this. Installing linux-headers seems to have partially solved the problem.

There is still a glitch. I tested this by uninstalling both tuxedo-cc-wmi and tuxedo-cc-wmi - having linux-headers also triggered dkms for all relevant kernels.
Then I rebooted (kernel 5.9) and noticed that while the CPU speed was detected in the control center the temperature and fan were still at 0° and 0% (sensors command reporting realistic temperatures of 40+° C). So I rebooted and everything seemed to work fine. I even rebooted into another two kernels (5.4 and 5.9-rt) and everything seemed to work.

To double-check I again removed the packages, rebooted, re-installed, re-booted. Again 0° and 0% fan. Tried rebooting again at least 3 times but still same problem, tried re-booting another kernel (5.4), same problem...
I tried unplugging and re-plugging in the mains. The profile was changing, but still 0/0 detected.
Then... I unplugged the mains _and_ with the mains unplugged rebooted, and everything was working.

So it seems that (maybe if there was some existing configuration), after (re)install the profile and / or monitoring of temperature and fan are not working until there is a change of mains / battery state AND a reboot, like the combination of plug/unplug + reboot 'trigger' something?

My 'test' for this is to do a 60 or 120 seconds with the stress -c 10 -t 120 command and monitor the control center as well as a looping sensors command and either top or htop in parallel both with mains and on battery.

Hope this is useful

**Comment by tuxedoxt** [comment]
Hello, if `dkms status` says tuxedo-cc-wmi is *installed*, have a look at this issue https://github.com/tuxedocomputers/tuxedo-control-center/issues/55 in case it's relevant.

**Comment by jmakov** [comment]
Same issue here on an old i7 Clevo laptop. The sensors command shows CPU temps, Control Center 0°C. Had worked before the latest update.

**Comment by tuxedoxt** [comment]
Hello, this.. is a rather outdated issue to post in. If this means you are using the old tuxedo-cc-wmi package you need to deinstall that one and install tuxedo-keyboard. It has the `tuxedo_io` module which is needed nowadays built-in.

Otherwise please make a new issue and post output of `dkms status` and `uname -r`.
